homeassistant/components/ws66i/__init__.py

```diff
@@ -0,0 +1,59 @@
+"""The Soundavo WS66i 6-Zone Amplifier integration."""
+import logging
+
+from pyws66i import get_ws66i
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_SOURCES, DOMAIN, WS66I_OBJECT
+
+PLATFORMS = ["media_player"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Soundavo WS66i 6-Zone Amplifier from a config entry."""
+    # Copy the config entry data to options for future use
+    if not entry.options:
+        options = {CONF_SOURCES: entry.data.get(CONF_SOURCES)}
+        hass.config_entries.async_update_entry(entry, options=options)
+
+    entry.async_on_unload(entry.add_update_listener(_update_listener))
+
+    ws66i = get_ws66i(entry.data[CONF_IP_ADDRESS])
+    try:
+        await hass.async_add_executor_job(ws66i.open)
+    except ConnectionError:
+        # Amplifier is probably turned off
+        _LOGGER.warning("Could not connect to WS66i Amp. Is it off?")
```

Comment:
Raise ConfigEntryNotReady(msg) instead

Reply:
Why should ConfigEntryNotReady be raised here? When should I return False in async_setup_entry()?

The name of the exception to me sounds like an exception that implies the information in the config entry is not ready to be used. Is that thought right?

Reply:
It will cause the integration to retry setup later in the hopes the device has been turned on. Otherwise a restart will be required to get it back up and going